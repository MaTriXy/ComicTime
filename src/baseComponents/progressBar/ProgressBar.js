import React from "react";
import PropTypes from "prop-types";
import { CircularProgress } from "material-ui";
import ComicTimeLogo from "../comicTimeLogo/ComicTimeLogo";

const style = {
  progressBar: {
    position: "absolute",
    left: "50%",
    top: "50%",
    transform: "translate(-50%, -50%)"
  },
  logo: {
    position: "absolute",
    width: 92,
    left: "50%",
    top: "50%",
    transform: "translate(-50%, -48%)"
  }
};
class ProgressBar extends React.Component {
  render() {
    if (this.props.hidden) {
      return null;
    } else {
      return (
        <div>
          <div>
            <CircularProgress
              mode="indeterminate"
              size={100}
              value={55}
              style={style.progressBar}
            />
            <div style={style.logo}>
              <ComicTimeLogo />
            </div>
          </div>
        </div>
      );
    }
  }
}

ProgressBar.propTypes = {
  hidden: PropTypes.bool
};

ProgressBar.defaultProps = {
  hidden: false
};

export default ProgressBar;
