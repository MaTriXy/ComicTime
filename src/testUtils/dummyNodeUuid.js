jest.mock("node-uuid", () => {
  return {
    v4() {
      return "uuid";
    }
  };
});
