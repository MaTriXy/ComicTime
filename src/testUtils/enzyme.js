import React from "react";
import Enzyme from "enzyme";
import { createSerializer } from "enzyme-to-json";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { MuiThemeProvider } from "material-ui";
import comicTimeTheme from "../theme";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
Enzyme.configure({ adapter: new Adapter() });

export function mockHistory() {
  return {
    go: jest.fn(),
    goBack: jest.fn()
  };
}

export function translator(key) {
  return key;
}

export function mountWithMuyThemeProvider(component) {
  return Enzyme.mount(
    <MuiThemeProvider muiTheme={comicTimeTheme}>{component}</MuiThemeProvider>
  );
}

export function mountComponent(component) {
  return Enzyme.mount(component);
}

export function mountComponentAsJson(component) {
  return toJson(mountComponent(component));
}

export function shallowComponent(component) {
  return Enzyme.shallow(component);
}

export function shallowComponentAsJson(component) {
  return toJson(shallowComponent(component));
}

export function json(component) {
  return toJson(component);
}
