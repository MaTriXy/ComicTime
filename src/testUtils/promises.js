export async function flushPromises() {
  return new Promise(resolve => setImmediate(resolve));
}
