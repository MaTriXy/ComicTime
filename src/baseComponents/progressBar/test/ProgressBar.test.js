jest.mock("../../../i18n/i18n");
jest.mock("material-ui/CircularProgress", () => "CircularProgress");
import React from "react";
import ProgressBar from "../ProgressBar";
import { shallowComponentAsJson } from "../../../testUtils/enzyme";

describe("ProgressBar", () => {
  it("renders the progress bar composed of a circular progress bar plus the app logo by default", () => {
    const component = shallowComponentAsJson(<ProgressBar />);
    expect(component).toMatchSnapshot();
  });

  it("renders the progress bar composed as by default if it is not hidden", () => {
    const component = shallowComponentAsJson(<ProgressBar hidden={false} />);
    expect(component).toMatchSnapshot();
  });

  it("renders an empty progress bar if it is hidden", () => {
    const component = shallowComponentAsJson(<ProgressBar hidden={true} />);
    expect(component).toMatchSnapshot();
  });
});
