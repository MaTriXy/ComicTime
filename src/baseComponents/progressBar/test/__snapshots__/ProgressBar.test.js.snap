// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders an empty progress bar if it is hidden 1`] = `""`;

exports[`ProgressBar renders the progress bar composed as by default if it is not hidden 1`] = `
<div>
  <div>
    <CircularProgress
      mode="indeterminate"
      size={100}
      style={
        Object {
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
        }
      }
      value={55}
    />
    <div
      style={
        Object {
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -48%)",
          "width": 92,
        }
      }
    >
      <ComicTimeLogo />
    </div>
  </div>
</div>
`;

exports[`ProgressBar renders the progress bar composed of a circular progress bar plus the app logo by default 1`] = `
<div>
  <div>
    <CircularProgress
      mode="indeterminate"
      size={100}
      style={
        Object {
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
        }
      }
      value={55}
    />
    <div
      style={
        Object {
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -48%)",
          "width": 92,
        }
      }
    >
      <ComicTimeLogo />
    </div>
  </div>
</div>
`;
